src/electron/ipc/ipc-message-dispatcher.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TelemetryData } from 'common/extension-telemetry-events';
+import { Message } from 'common/message';
+import { ActionMessageDispatcher } from 'common/message-creators/types/dispatcher';
+import { IPC_MESSAGE_CHANNEL_NAME } from './ipc-channel-names';
+
+// The most common type of sink is BrowserWindow.webContext.send
+export type IpcMessageSink = (channelName: string, message: Message) => void;
+
+export class IpcMessageDispatcher implements ActionMessageDispatcher {
+    public dispatchMessage(message: Message): void {
+        if (this.registeredSinks.length === 0) {
+            this.messageBacklog.push(message);
+            return;
+        }
+
+        for (const sink of this.registeredSinks) {
+            sink(IPC_MESSAGE_CHANNEL_NAME, message);
+        }
+    }
+
+    public dispatchType(messageType: string): void {
+        throw new Error('Method not implemented (not yet required for any main process events).');
+    }
+    public sendTelemetry(eventName: string, eventData: TelemetryData): void {
+        throw new Error('Method not implemented (not yet required for any main process events).');
+    }
+
+    private messageBacklog: Message[] = [];
```

Comment:
It's odd that we have instance properties in the middle of the class. Should they be on the top?